src/java.desktop/share/classes/com/sun/java/swing/plaf/gtk/GTKLookAndFeel.java

```diff
@@ -803,7 +803,7 @@ public Object createValue(UIDefaults table) {
             "PasswordField.font", new FontLazyValue(Region.PASSWORD_FIELD),
 
 
-            "PopupMenu.consumeEventOnClose", Boolean.TRUE,
+            "PopupMenu.consumeEventOnClose", Boolean.FALSE,
```

Comment:
This property was added to support some kind of "native" behavior.
The code from the BasicPopupMenuUI.java:

```
                    // Ask UIManager about should we consume event that closes
                    // popup. This made to match native apps behaviour.
                    boolean consumeEvent =
                        UIManager.getBoolean("PopupMenu.consumeEventOnClose");
                    // Consume the event so that normal processing stops.
                    if(consumeEvent && !(src instanceof MenuElement)) {
                        me.consume();
                    }
```
So after this fix, the mouse event that causes the popup to close will be not be dispatched to the next component?

Reply:
Before the fix the mouse event that cause the popup to close  was consumed here as the "PopupMenu.consumeEventOnClose" was true.

After the fix the mouse event that cause the popup to close will not be consumed here as  "PopupMenu.consumeEventOnClose" is set to false in the fix for Windows, GTK, Nimbus and Motif LAF.

Reply:
And does it match the native behavior? I mean different values of "consumeEventOnClose" weren't a bug. It was intentionally set to the appropriate value.

Reply:
yeah , i have checked in windows and ubuntu native apps and it is matching native behaviour.
The issue is seen in Motif, Nimbus, GTK and windows which sets consumeEventOnClose to true.
It is not seen in Metal And Aqua which doesn't set this variable and the value from BasicLookAndFeel.java is used which is false.

Reply:
Can you recheck that using the next usecase:
 - Create menu in the frame
 - Add jbutton to the frame exactly in the place where the menu popup is opened
 - Open menu and click some menu item, will the jbutton be clicked?

Reply:
Open menu and click some menu item, will the jbutton be clicked?
    No, the JButton is not clicked in this use case.